core/shell.py

```diff
@@ -548,8 +548,28 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
   assert home_dir is not None
   rc_path = flag.rcfile
   # mycpp: rewrite of or
+
+  rc_paths = []
+  # TODO: Decide if this should be first or last rc_path (as in: should it overwrite modular includes or not)
```

Comment:
I'm thinking it should be the last one, so the user can REMOVE stuff that  nix and starship added if they want?  Without editing their code

This is related to the "Neighborly Namespacing Idea" and `trap --add` and `trap --remove`


https://oilshell.zulipchat.com/#narrow/stream/121540-oil-discuss/topic/Neighborly.20Namespacing.20Idea

Reply:
> Actually I'm thinking the user RC file should be the first one, and then we add an EVENT `after-rcdir` that lets the user run code there? LIke this
> 
> ```
> trap --event after-rcdir {
>   echo 'do other stuff'
> }
> ```
> 
> That can be done later, but I think giving the user the "first shot" makes sense

Yeah exactly. That would also allow easy "preconfiguration" for plugins. Plus it would theoretically be possible to change things at the end with a file like `ZZZ_replace_xyz.sh` (though a `trap --event after-rcdir` as you suggested is much cleaner!).